File: Squares of a Sorted Array.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        int i = 0;
        int j = nums.size() - 1;
        vector<int> arr;
        arr.reserve(nums.size());
        int sq_i = nums.at(i)*nums.at(i);;
        int sq_j = nums.at(j)*nums.at(j);;

        while (i < j) {
            if (sq_i > sq_j) {
                arr.push_back(sq_i);
                i++;
                sq_i = nums.at(i)*nums.at(i);
            } else {
                arr.push_back(sq_j);
                j--;
                sq_j = nums.at(j)*nums.at(j);
            }
        }

        arr.push_back(sq_i);
        i = 0;
        j = arr.size() - 1;
        while (i < j) {
            swap(arr.at(i++), arr.at(j--));
        }
        return arr;
    }
};

File: Squares of a Sorted Array.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        i, j = 0, len(nums) - 1
        arr = []
        sq_i, sq_j = nums[i] * nums[i], nums[j] * nums[j]
        while i < j:
            if sq_i > sq_j:
                arr.append(sq_i)
                i += 1
                sq_i = nums[i] * nums[i]
            else:
                arr.append(sq_j)
                j -= 1
                sq_j = nums[j] * nums[j]

        arr.append(sq_i)
        return arr[::-1]
